server/npc.ts

```diff
@@ -0,0 +1,44 @@
+import { Direction } from "../shared/messages.js";
+import { GameState } from "../shared/state.js";
+
+import { getRandomElement, isBeachTile } from "./utils.js";
+
+export default class NPC {
+  // hack, copied over to make this a valid ServerPlayer since depending on it
+  // would make a circular dependency--maybe should clean up later
+  id: string;
+  x: number;
+  y: number;
+  direction: Direction;
+  isNpc: true = true;
```

Comment:
would like to clean this up, we should maybe have a base interface that both `NPCPlayer` and `ServerPlayer` implement

Reply:
Yeah I thought about that but what functions would the non-npc player have?

Reply:
they could both have a generic `update()` method?

Reply:
that's true but they would both types would have the same method which would be applying the game speed, so that i would put that in the abstract class and then have an empty implementation for the non-npc player

if you prefer that i'm happy to do that

Reply:
ok that sounds better to me